sw/eventListener.js

```diff
@@ -155,5 +167,18 @@ export function onMessage (sw) {
     if (event.data.action === 'SYNC_SUBSCRIPTION') {
       return event.waitUntil(onPushSubscriptionChange(sw)(event.oldSubscription, event.newSubscription))
     }
+    if (event.data.action === CLEAR_BADGE_ACTION) {
+      return event.waitUntil((async () => {
+        let notifications = []
+        try {
+          notifications = await sw.registration.getNotifications()
+        } catch (err) {
+          console.error('failed to get notifications')
```

Comment:
Missing return? In case of error, you probably don't want to access `notifications`

Reply:
I initialized `notifications` to `[]` to handle this case. I figured even if it failed to fetch notifications, we can clear the badge, but perhaps that's a bad idea.